### ISSUE TITLE
Clarification on Light implementation

### ISSUE THREAD

**Issue by ginkxo** [dilevin/computer-graphics-ray-tracing#19]
Regarding directional light vs. point light: 

- We have to calculate vector direction "d" pointing to the direction of the light source just from the information in the query "q"?
- does "q" as a query point refer to the actual position of an object being hit by a ray in 3d space, or something else?
- how exactly do we interpret max_t here? if max_t is infinite, then it means the light source is extremely far away, but if it is a value say < 1.0 the light source position d is very close? and max_t is something that needs to be calculated based off of just object position as well?

thanks!

**Comment by rarora7777** [comment]
I'd give a hint here since some of you may be new to C++ or Object-Oriented Programming:

The function `PointLight::direction` (or `DirectionalLight::direction`) is a **member function** of the `PointLight` (or `DirectionalLight`) class. So, you also have access to the **member variables** (also known as fields) of the object for which the function is called. This means that you also have access to properties of that light, and not just the parameter (`q`) explicitly passed to the function.

Look at `PointLight.h` (or `DirectionalLight.h`) to see what properties are available to you.

**Comment by rarora7777** [comment]
`max_t` is the parametric distance of the query point `q` from the light source. For a point light, this will be the value of the parameter `t` along the light ray. For a directional light, the `README` tells you precisely what value you need to set here. 😉

**Comment by radiolariax** [comment]
right, I was getting confused over how the d field in specifically DirectionalLight was both being used as a "direction from the light towards the scene" but we also return d as a "direction towards the light" in our .direction implementations. I'm assuming now that since PointLight has its member variable called p that this was just a case of variable naming I accidentally overlooked. thanks for the help!
